src/System.ServiceModel.Syndication/src/System/ServiceModel/Syndication/XmlWriterWrapper.cs

```diff
@@ -249,23 +153,92 @@ public static Task WriteElementStringAsync(this XmlWriter writer, string localNa
 
         public static Task WriteAttributeStringAsync(this XmlWriter writer, string localName, string ns, string value)
         {
-            return writer.InternalWriteAttributeStringAsync(null, localName, ns, value);
+            return writer.WriteAttributeStringAsync(null, localName, ns, value);
         }
 
         public static Task WriteAttributeStringAsync(this XmlWriter writer, string localName, string value)
         {
-            return writer.InternalWriteAttributeStringAsync(null, localName, null, value);
+            return writer.WriteAttributeStringAsync(null, localName, null, value);
         }
 
-        public static Task InternalWriteAttributeStringAsync(this XmlWriter writer, string prefix, string localName, string ns, string value)
+        // XmlDictionaryWriter.WriteAsync isn't supported for the type of writer returned by XmlDictionaryWriter.CreateBinaryWriter
+        // as that writer isn't Async. We still want to read from the reader async so this is a hybrid implementation of WriteNodeAsync.
+        // It uses synchronous api's when using the writer and async api's when using the reader.
+        public static async Task InternalWriteNodeAsync(this XmlDictionaryWriter writer, XmlReader reader, bool defattr)
         {
-            if (writer is XmlWriterWrapper)
+            char[] writeNodeBuffer = null;
+            const int WriteNodeBufferSize = 1024;
+
+            if (null == reader)
             {
-                writer.WriteAttributeString(prefix, localName, ns, value);
-                return Task.CompletedTask;
+                throw new ArgumentNullException(nameof(reader));
             }
 
-            return writer.WriteAttributeStringAsync(prefix, localName, ns, value);
+            bool canReadChunk = reader.CanReadValueChunk;
+            int d = reader.NodeType == XmlNodeType.None ? -1 : reader.Depth;
+            do
+            {
+                switch (reader.NodeType)
+                {
+                    case XmlNodeType.Element:
+                        writer.WriteStartElement(reader.Prefix, reader.LocalName, reader.NamespaceURI);
+                        writer.WriteAttributes(reader, defattr);
+                        if (reader.IsEmptyElement)
+                        {
+                            writer.WriteEndElement();
+                            break;
```

Comment:
The break is not necessary.

Reply:
Fixed. Although this code was actually copied from the full framework with a couple of sync methods replaced with their async variant!!